### PR TITLE
Fix documentation workflow block of PRs and add names to jobs

### PR DESCRIPTION
**Pull request by inf17101**
Issues: #71 

- moves the path filter of the workflow trigger to a job conditional (because job conditionals do not mark a GH workflow as pending or failed if they skip the job)

- adds names to the GH action jobs

# Definition of Done

The PR shall be merged **only if all of following items were checked.** In case an item is not applicable as described, please provide a short explanation right after the item in question. Separate the explanation by a semicolon and write it as bold text like **; requirements are handled in issue #xyz**:

- [ ] documentation of all modules `/*/doc/swdesign` is up-to-date
- [ ] requirements are up-to-date and requirements for new features have been added and mapped to source and test
- [ ] conform to [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/)

### CODE REVIEW

File: doc/docs/development/ci-cd.md [hunk 1]
```diff
@@ -58,15 +58,35 @@ whenever the authors of the Github action update the generic tag to point to a n
...
+
+jobs:
+  test_and_build_linux_amd64:
+    name: Test and Build Linux amd64
+...
+```
+
+!!! Note
+
+    Giving a job a name is a best practice and if a job needs to configured as required status check inside the [self-service repository](https://github.com/eclipse-ankaios/.eclipsefdn) then this job is referenced by its name inside the configuration.
```
Comment (krucod3) [inline doc/docs/development/ci-cd.md:90]:
```suggestion
    Beside being a best practice, giving a job a name is needed to reference it from the [self-service repository](https://github.com/eclipse-ankaios/.eclipsefdn) in order to configure the job as a required status check.
```

**Review by krucod3 — COMMENTED** [review]
Reviewed. Documented a small improvement idea.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -12,14 +12,15 @@ env:
 
 jobs:      
   test_and_build_linux_amd64:
+    name: Test and Build Linux amd64
```
Comment (windsource) [inline .github/workflows/build.yml:15]:
```suggestion
    name: Test and build Linux amd64
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -92,6 +94,7 @@ jobs:
           path: target/aarch64-unknown-linux-gnu/debian/ankaios*.deb
 
   requirements:
+    name: Build Requirements Tracing
```
Comment (windsource) [inline .github/workflows/build.yml:97]:
```suggestion
    name: Build requirements tracing
```

File: .github/workflows/documentation.yml [hunk 4]
```diff
@@ -3,28 +3,39 @@ on:
...
-      - 'api/proto/**'
-      - '.github/workflows/documentation.yml'
   workflow_dispatch:
   workflow_call:
 
 permissions:
   contents: write
 jobs:
+  documentation_changes:
+    name: Check If Documentation Changed
```
Comment (windsource) [inline .github/workflows/documentation.yml:14]:
```suggestion
    name: Check if documentation has changed
```

File: .github/workflows/documentation.yml [hunk 5]
```diff
@@ -3,28 +3,39 @@ on:
...
+    - uses: dorny/paths-filter@v2.11.1
+      id: filter
+      with:
+        filters: |
+          documentation:
+              - 'doc/**'
+              - 'api/proto/**'
+              - '.github/workflows/documentation.yml'
   deploy:
+    name: Deploy Documentation
```
Comment (windsource) [inline .github/workflows/documentation.yml:31]:
```suggestion
    name: Deploy documentation
```

File: .github/workflows/documentation.yml [hunk 6]
```diff
@@ -3,28 +3,39 @@ on:
...
+    needs: documentation_changes
+    if: ${{ needs.documentation_changes.outputs.documentation == 'true' || github.ref_name == 'main' || github.ref_type == 'tag' }}
```
Comment (windsource) [inline .github/workflows/documentation.yml:33]: But that means, that the new path filtering is only active for PRs and not for main, right?

File: doc/docs/development/self-service.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+If a configuration needs to be changed the process is the following:
+
+1. Fork the [.eclipsefdn](https://github.com/eclipse-ankaios/.eclipsefdn) repository.
+2. Do the configuration changes (Use the [Eclipse playground](https://eclipse-ankaios.github.io/.eclipsefdn/playground/) for trying out the available settings).
+3. Open a PR pointing from your fork's branch to the [.eclipsefdn](https://github.com/eclipse-ankaios/.eclipsefdn) repository.
+4. Request review from: Ankaios project committer, eclipsefdn-releng, eclipsefdn-security.
+5. After the changes were approved by the reviewers, merge the changes into the main branch.
+
+After the changes were merged an Eclipse member will apply the new settings by using the [otterdog cli](https://otterdog.readthedocs.io/en/latest/).
```
Comment (windsource) [inline doc/docs/development/self-service.md:22]: Does this mean that although we have a self service, after merging the PR a manual step is required by Eclipse staff?

**Review by windsource — CHANGES_REQUESTED** [review]
Some minor changes requested and some questions.

File: .github/workflows/documentation.yml (hunk 5 above)
Comment (inf17101) [reply .github/workflows/documentation.yml:31]: Fix name of deploy doc job

File: .github/workflows/documentation.yml (hunk 6 above)
Comment (inf17101) [reply .github/workflows/documentation.yml:33]: Yes the path filtering is active for PRs and skipped for pushes on main and releases. I have kept the conditional simple and weak to reduce complexity. For me it was important that it not runs on every commit in a PR. But if it runs on every push on main, it is not too bad because merging a PR happens quite less often then a committing into a PR.

We can make it stricter like this I think (not tested):
`if: ${{ needs.documentation_changes.outputs.documentation == 'true' || github.ref_name == 'main' && needs.documentation_changes.outputs.documentation == 'true' || github.ref_type == 'tag' }}`

File: doc/docs/development/self-service.md (hunk 7 above)
Comment (inf17101) [reply doc/docs/development/self-service.md:22]: Yes, this is how it is currently. But they have written that they are working on automating such things into CI / CD, but this needs still a little bit of testing for now.

File: .github/workflows/documentation.yml (hunk 6 above)
Comment (windsource) [reply .github/workflows/documentation.yml:33]: OK I see. Then the change like in the original PR is OK for me.

File: doc/docs/development/self-service.md (hunk 7 above)
Comment (windsource) [reply doc/docs/development/self-service.md:22]: OK. But then we need rephrase it to:

```suggestion
After the changes were merged, a member of Eclipse staff will apply the new settings by using the [otterdog cli](https://otterdog.readthedocs.io/en/latest/).
```

because 'member of Eclipse' are all member companies.

File: doc/docs/development/self-service.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+A detailed overview of the self-service please have a look into the [self-service handbook](https://www.eclipse.org/projects/handbook/#resources-github-self-service).
+
+## Process of changing the settings
+
+If a configuration needs to be changed the process is the following:
+
+1. Fork the [.eclipsefdn](https://github.com/eclipse-ankaios/.eclipsefdn) repository.
+2. Do the configuration changes (Use the [Eclipse playground](https://eclipse-ankaios.github.io/.eclipsefdn/playground/) for trying out the available settings).
+3. Open a PR pointing from your fork's branch to the [.eclipsefdn](https://github.com/eclipse-ankaios/.eclipsefdn) repository.
+4. Make sure that review is request from: Ankaios project committer, eclipsefdn-releng, eclipsefdn-security.
```
Comment (windsource) [inline doc/docs/development/self-service.md:19]:
```suggestion
4. Make sure that a review is requested from: Ankaios project committer, eclipsefdn-releng, eclipsefdn-security.
```

**Review by windsource — APPROVED** [review]
LGTM
